src/main/java/net/openhft/chronicle/wire/WireOut.java

```diff
@@ -193,10 +193,25 @@ default void writeNotCompleteDocument(boolean metaData, @NotNull WriteMarshallab
      * @param timeout      throw TimeoutException if it could not write the marker in time.
      * @param timeUnit     of the timeout
      * @param lastPosition the end of the wire
-     * @return did this method write EOF or was it already there.
+     * @return {code true} if did this method wrote EOF, {@code false} if it was already there.
      */
     boolean writeEndOfWire(long timeout, TimeUnit timeUnit, long lastPosition);
 
+    /**
+     * Check if end of wire marker is not present, optionally writing it unless one is already written.
+     * This will increment the headerNumber as appropriate if successful
+     *
+     * @param writeEOF     if {@code true}, write end of wire marker unless already exists
+     * @param timeout      throw TimeoutException if it could not write the marker in time.
+     * @param timeUnit     of the timeout
+     * @param lastPosition the end of the wire
+     * @return {@code true} if did this method could write an EOF (depending on write parameter).
+     *         {@code false} if it was it already there.
+     */
+    default boolean noEndOfWire(boolean writeEOF, long timeout, TimeUnit timeUnit, long lastPosition) {
```

Comment:
what about if we called it `isEndOfWirePresent` and the `writeEOF` parameter could be called `writeIfNotPresent`

Reply:
ohh the return value is weird. I guess to support writeEOF

Reply:
I don't like the method signature, it requires a lot of mental gymnastics to understand.

Reply:
Perhaps we should return an enum: e.g. `(NotPresent|PresentAfterUpdate|Present)`

Reply:
So to clarify... we return false if the EOF is already there, and true if it's not there. But if `writeEOF` was true and we returned true it means we wrote EOF.

I think it would be clearer if the method were called `isEndOfWirePresent(boolean writeEOF, ...)`
And it returned `(NotPresent|Present|PresentAfterUpdate)`

i.e.
|State of EOF|writeEOF|existing return value|proposed return value|
|------------------|-------------|------------------------------|---------------------------------|
|present|true|false|Present|
|present|false|false|Present|
|not present|true|true|PresentAfterUpdate|
|not present|false|true|NotPresent|

Reply:
Then writeEOF could be imeplemented as
```
    @Override
    public boolean writeEndOfWire(long timeout, TimeUnit timeUnit, long lastPosition) {
        return isEndOfWirePresent(true, timeout, timeUnit, lastPosition) == PresentAfterUpdate;
    }
```

Reply:
Or even better why not just have writeEndOfWire which will write it, and isEndOfWirePresent which will ONLY read it?

Reply:
Having two of them will be problematic since two subsequent invocations can produce irrelevant results. It will also make it much harder to implement. I'll go with enum.